From a670036eb522235012e57be4c38a9f3882e7e8ac Mon Sep 17 00:00:00 2001
From: "Tao, Jing" <jing.tao@intel.com>
Date: Wed, 24 Dec 2014 15:17:33 +0800
Subject: [PATCH 195/196] ov5693-ecs: WA tunning for 1424x1168 resolution

This is the patch to workaround the firmware downscaling
issue.

Refer: https://hsdes.intel.com/home/default.html#article?id=1204818921

Change-Id: Ic29fb5f0ac94440b45d52443ccecc9523d2fbf59
Tracked-On: https://hsdes.intel.com/home/default.html
	#article?id=1204720930
Signed-off-by: Tao, Jing <jing.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/34089
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h   | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
index 25078a0..248e632 100755
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
@@ -808,6 +808,8 @@ static struct ov5693_reg const ov5693_2592x1944_30fps[] = {
  * 11:9 Full FOV Output, expected FOV Res: 2346x1920
  * ISP Effect Res: 1408x1152
  * Sensor out: 1424x1168, DS From: 2380x1952
+ *
+ * WA: Left Offset: 8, Hor scal: 64
  */
 static struct ov5693_reg const ov5693_1424x1168_30fps[] = {
 	{OV5693_8BIT, 0x3501, 0x3b}, /* long exposure[15:8] */
@@ -815,11 +817,11 @@ static struct ov5693_reg const ov5693_1424x1168_30fps[] = {
 	{OV5693_8BIT, 0x3708, 0xe2},
 	{OV5693_8BIT, 0x3709, 0xc3},
 	{OV5693_8BIT, 0x3800, 0x00}, /* TIMING_X_ADDR_START */
-	{OV5693_8BIT, 0x3801, 0x78}, /* 110 */
+	{OV5693_8BIT, 0x3801, 0x50}, /* 80 */
 	{OV5693_8BIT, 0x3802, 0x00}, /* TIMING_Y_ADDR_START */
 	{OV5693_8BIT, 0x3803, 0x02}, /* 2 */
 	{OV5693_8BIT, 0x3804, 0x09}, /* TIMING_X_ADDR_END */
-	{OV5693_8BIT, 0x3805, 0xc5}, /* 2501 */
+	{OV5693_8BIT, 0x3805, 0xdd}, /* 2525 */
 	{OV5693_8BIT, 0x3806, 0x07}, /* TIMING_Y_ADDR_END */
 	{OV5693_8BIT, 0x3807, 0xa1}, /* 1953 */
 	{OV5693_8BIT, 0x3808, 0x05}, /* TIMING_X_OUTPUT_SIZE */
-- 
1.9.1

